woocommerce/woocommerce-ios
# [Mobile Payments] [Several Readers Found] Integrate choose-a-reader list modal

Partially addresses #4333 

Changes:
- Integrates the `SeveralReadersFoundViewController` into `CardReaderSettingsAlerts`
- Disables the inter-modal animations from the previous PR to be less jumpy when transitioning from single-reader-found to several - cc @Garance91540 @adamzelinski - thoughts? (see movie below)
- Adds a new state to `CardReaderConnectionController` for when multiple readers have been found
- Adds a new flag to `CardReaderConnectionController` to avoid switching out of the list view if the number of found readers drops below 2 during a particular searching flow
- Integrates the new `foundSeveralReaders` alert into `CardReaderConnectionController`
- Connects the connect buttons and cancel button in `SeveralReadersFoundViewController` to do their work
- Updates the `ActivitySpinnerAndLabelTableViewCell` to keep it spinning after re-use

What's next:
- The next (and last) PR for this issue will remove the feature flag and launch the feature for all users

To Test:

- First, set simulator to true in your Run Scheme and then run the app
- Tap on the gear icon, then In-Person Payments, then Manage card reader
- If you have a known reader, search will begin automatically. If not, tap the Connect Card Reader button
- A searching modal should display, possibly quite briefly
- You’ll then get a “Do you want to connect to reader CHB...” modal. Tap on Keep Searching.
- You’ll get the “Several readers found” list modal.
- Tap the cancel button and ensure it cancels search, and that you go back to the “Connect your card reader” screen and that nothing further happens (e.g. wait for at least 10 seconds)
- Tap the Connect Card Reader button to initiate another search
- A searching modal should display, possibly quite briefly
- You’ll then get a “Do you want to connect to reader CHB...” modal. Tap on Keep Searching.
- You’ll get the “Several readers found” list modal.
- With the simulator, you may initially see two entries (one for CHB20....1 and one for CHB20...2) but that quickly will just show the CHB20...2 reader
- Tap on Connect on the CHB20...2 reader’s row
- Ensure you are connected to the CHB20...2 reader
- Next, repeat the tests but with multiple physical readers
- Next, repeat the tests with the collect-payment flow (be sure to disconnect from each known reader in settings first if you want to avoid automatic known reader connection)
- Next, disable the feature flag, and retest ensuring that multiple found readers are presented one at a time during discovery (to disable the feature flag easily, add a return at the beginning of updateShowSeveralFoundReaders)
- Lastly, ensure all unit tests pass

Screengrabs / movies:

@Garance91540 : I have updated both of these movies today, Sep 10. In both cases I had to tap the "Keep Searching" button to continue on to the several-readers-found list view.

https://user-images.githubusercontent.com/1595739/132927754-9d7a87f2-9655-4b9d-a10d-a43699e9ac78.MP4


https://user-images.githubusercontent.com/1595739/132927778-072a6982-516d-404b-a89a-bf64f16ec0e8.MP4

Update release notes:
